2.66 crashes, not sure on which resource

```
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00145c8c0, 0x655b5c8, 0xc000b7ecc0, 0xc0043a2a80, 0x560bc60, 0xc000ff7c00, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:335 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00145c8c0, 0x655b5c8, 0xc000b7ecc0, 0xc00149a930, 0x560bc60, 0xc000ff7c00, 0xc0040aa678, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:624 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0014ee0a8, 0x655b5c8, 0xc000b7ecc0, 0xc000b7ed00, 0xc000b7ecc0, 0x40ba05, 0x591b140)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:575 +0x43b
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc000b5e0e0, 0x655b670, 0xc000b7ecc0, 0xc0017237a0, 0xc000b5e0e0, 0xc003a5b140, 0xc001e77ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:298 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x5ad47c0, 0xc000b5e0e0, 0x655b670, 0xc003a5b140, 0xc001723740, 0x0, 0x655b670, 0xc003a5b140, 0xc003e40380, 0x374)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00013f6c0, 0x65a4dd8, 0xc000103080, 0xc003b0e000, 0xc0004a6510, 0x9a7ab30, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00013f6c0, 0x65a4dd8, 0xc000103080, 0xc003b0e000, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0011417a4, 0xc00013f6c0, 0x65a4dd8, 0xc000103080, 0xc003b0e000)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd
Error: The terraform-provider-azurerm_v2.66.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

+1

Error: rpc error: code = Unavailable desc = transport is closing


panic: reflect: call of reflect.Value.IsNil on array Value
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: goroutine 1040 [running]:
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: reflect.Value.IsNil(...)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/goenv/versions/1.16.3/src/reflect/value.go:1086
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/mitchellh/mapstructure.(*Decoder).decodeSlice(0xc0017b11e8, 0x0, 0x0, 0x5513a40, 0xc002906830, 0x546ed40, 0xc00105dfb0, 0x197, 0xc0017b11e8, 0x8)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/mitchellh/mapstructure/mapstructure.go:1086 +0x10ad
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/mitchellh/mapstructure.(*Decoder).decode(0xc0017b11e8, 0x0, 0x0, 0x5513a40, 0xc002906830, 0x546ed40, 0xc00105dfb0, 0x197, 0xc00105dfb0, 0x197)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/mitchellh/mapstructure/mapstructure.go:460 +0x836
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/mitchellh/mapstructure.(*Decoder).Decode(0xc0017b11e8, 0x5513a40, 0xc002906830, 0x0, 0xc00105dfb0)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/mitchellh/mapstructure/mapstructure.go:389 +0xf0
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/mitchellh/mapstructure.Decode(0x5513a40, 0xc002906830, 0x53e0020, 0xc00105dfb0, 0x1, 0xc000bf4b50)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/mitchellh/mapstructure/mapstructure.go:297 +0xa5
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setList(0xc00105dde8, 0xc0020affb0, 0x3, 0x3, 0x5513a40, 0xc002906830, 0xc00102a640, 0xc0029087f8, 0xffffffffffffffff)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:127 +0xd2
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc00105dde8, 0xc0020affb0, 0x3, 0x3, 0x5513a40, 0xc002906830, 0x0, 0x0)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:103 +0xde
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setObject(0xc00105dde8, 0xc001aecca0, 0x2, 0x2, 0x563cd60, 0xc0020afda0, 0xc00290a000, 0x10, 0x60122f0)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:225 +0x21a
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc00105dde8, 0xc001aecca0, 0x2, 0x2, 0x563cd60, 0xc0020afda0, 0x4, 0xaf9559)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:109 +0x1de
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setList.func1(...)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:123
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setList(0xc00105dde8, 0xc0029068f0, 0x1, 0x1, 0x546ed40, 0xc00105df38, 0xc00102b7c0, 0x5e04135, 0xc001aef900)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:143 +0x29e
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc00105dde8, 0xc0029068f0, 0x1, 0x1, 0x546ed40, 0xc00105df38, 0x1, 0x1)
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:103 +0xde
2021-07-05T08:00:58.711Z [DEBUG] plugin.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).WriteField(0xc00105dde8, 0xc0029068f0, 0x1, 0x1, 0x546ed40, 0xc00105df38, 0x0, 0x0)

Hello,

Same crash for me with v2.66. If I use the previous version (v2.65), everything works perfectly.

This is the message

```bash
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: PANIC: runtime error: invalid memory address or nil pointer dereference
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5b804ea]
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5:
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: goroutine 1740 [running]:
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/recoveryservices.resourceSiteRecoveryReplicatedItemRead(0xc002958300, 0x620a540, 0xc001221500, 0x0, 0x0)
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/recoveryservices/site_recovery_replicated_vm_resource.go:465 +0xb6a
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000efaa80, 0x715ab68, 0xc0016ea1c0, 0xc002958300, 0x620a540, 0xc001221500, 0x0, 0x0, 0x0)
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:335 +0x1ee
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000efaa80, 0x715ab68, 0xc0016ea1c0, 0xc0010941c0, 0x620a540, 0xc001221500, 0xc000866358, 0x0, 0x0, 0x0)
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:624 +0x1cb
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00135a738, 0x715ab68, 0xc0016ea1c0, 0xc0016ea200, 0xc0016ea1c0, 0x100b705, 0x6519e00)
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:575 +0x43b
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc0013af3a0, 0x715ac10, 0xc0016ea1c0, 0xc001516300, 0xc0013af3a0, 0xc003298720, 0xc002745ba0)
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:298 +0x105
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x66d3400, 0xc0013af3a0, 0x715ac10, 0xc003298720, 0xc0015162a0, 0x0, 0x715ac10, 0xc003298720, 0xc00254c400, 0xb16)
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001fe8c0, 0x71a4538, 0xc000602900, 0xc002496000, 0xc000b0a2d0, 0xa67dcd0, 0x0, 0x0, 0x0)
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc0001fe8c0, 0x71a4538, 0xc000602900, 0xc002496000, 0x0)
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000b172c0, 0xc0001fe8c0, 0x71a4538, 0xc000602900, 0xc002496000)
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-07-05T16:47:22.532+0200 [DEBUG] provider.terraform-provider-azurerm_v2.66.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd
2021-07-05T16:47:22.538+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.538+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.538+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.538+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.538+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.538+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.538+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.538+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.538+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.538+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.539+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-07-05T16:47:22.539+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/azurerm/2.66.0/darwin_amd64/terraform-provider-azurerm_v2.66.0_x5 pid=14826 error="exit status 2"
2021-07-05T16:47:22.539+0200 [ERROR] plugin.(*GRPCProvider).UpgradeResourceState: error="rpc error: code = Canceled desc = context canceled"
2021-07-05T16:47:22.540+0200 [INFO]  backend/local: plan operation completed
╷
│ Error: Plugin did not respond
│
│   with azurerm_site_recovery_replicated_vm.vm-replication-vm-ecom-002,
│   on 05-backups.tf line 154, in resource "azurerm_site_recovery_replicated_vm" "vm-replication-vm-ecom-002":
│  154: resource "azurerm_site_recovery_replicated_vm" "vm-replication-vm-ecom-002" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with azurerm_linux_virtual_machine_scale_set.scalesetlinuxsharenfs01,
│   on 10-deployshareVm.tf line 73, in resource "azurerm_linux_virtual_machine_scale_set" "scalesetlinuxsharenfs01":
│   73: resource "azurerm_linux_virtual_machine_scale_set" "scalesetlinuxsharenfs01" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with azurerm_monitor_autoscale_setting.autoscalefrontend01,
│   on 20-frontend.tf line 311, in resource "azurerm_monitor_autoscale_setting" "autoscalefrontend01":
│  311: resource "azurerm_monitor_autoscale_setting" "autoscalefrontend01" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with azurerm_monitor_autoscale_setting.autoscalefrontend02,
│   on 20-frontend.tf line 459, in resource "azurerm_monitor_autoscale_setting" "autoscalefrontend02":
│  459: resource "azurerm_monitor_autoscale_setting" "autoscalefrontend02" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with azurerm_monitor_autoscale_setting.autoscalebackend01,
│   on 30-backend.tf line 139, in resource "azurerm_monitor_autoscale_setting" "autoscalebackend01":
│  139: resource "azurerm_monitor_autoscale_setting" "autoscalebackend01" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with azurerm_virtual_machine_scale_set_extension.health_agent,
│   on 30-backend.tf line 196, in resource "azurerm_virtual_machine_scale_set_extension" "health_agent":
│  196: resource "azurerm_virtual_machine_scale_set_extension" "health_agent" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Request cancelled
│
│   with azurerm_monitor_autoscale_setting.autoscalebackend02,
│   on 30-backend.tf line 268, in resource "azurerm_monitor_autoscale_setting" "autoscalebackend02":
│  268: resource "azurerm_monitor_autoscale_setting" "autoscalebackend02" {
│
│ The plugin.(*GRPCProvider).UpgradeResourceState request was cancelled.
╵
╷
│ Error: Plugin did not respond
│
│   with azurerm_virtual_machine_scale_set_extension.health_agent02,
│   on 30-backend.tf line 325, in resource "azurerm_virtual_machine_scale_set_extension" "health_agent02":
│  325: resource "azurerm_virtual_machine_scale_set_extension" "health_agent02" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with azurerm_linux_virtual_machine_scale_set.scalesetlinuxstaging01[0],
│   on 50-staging.tf line 86, in resource "azurerm_linux_virtual_machine_scale_set" "scalesetlinuxstaging01":
│   86: resource "azurerm_linux_virtual_machine_scale_set" "scalesetlinuxstaging01" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with azurerm_mssql_server.sqlserver01,
│   on 60-sql-server.tf line 21, in resource "azurerm_mssql_server" "sqlserver01":
│   21: resource "azurerm_mssql_server" "sqlserver01" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with azurerm_mssql_server.sqlserver51,
│   on 60-sql-server.tf line 117, in resource "azurerm_mssql_server" "sqlserver51":
│  117: resource "azurerm_mssql_server" "sqlserver51" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-azurerm_v2.66.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5b804ea]

goroutine 1740 [running]:
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/recoveryservices.resourceSiteRecoveryReplicatedItemRead(0xc002958300, 0x620a540, 0xc001221500, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/recoveryservices/site_recovery_replicated_vm_resource.go:465 +0xb6a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000efaa80, 0x715ab68, 0xc0016ea1c0, 0xc002958300, 0x620a540, 0xc001221500, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:335 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000efaa80, 0x715ab68, 0xc0016ea1c0, 0xc0010941c0, 0x620a540, 0xc001221500, 0xc000866358, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:624 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00135a738, 0x715ab68, 0xc0016ea1c0, 0xc0016ea200, 0xc0016ea1c0, 0x100b705, 0x6519e00)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:575 +0x43b
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc0013af3a0, 0x715ac10, 0xc0016ea1c0, 0xc001516300, 0xc0013af3a0, 0xc003298720, 0xc002745ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:298 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x66d3400, 0xc0013af3a0, 0x715ac10, 0xc003298720, 0xc0015162a0, 0x0, 0x715ac10, 0xc003298720, 0xc00254c400, 0xb16)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001fe8c0, 0x71a4538, 0xc000602900, 0xc002496000, 0xc000b0a2d0, 0xa67dcd0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0001fe8c0, 0x71a4538, 0xc000602900, 0xc002496000, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000b172c0, 0xc0001fe8c0, 0x71a4538, 0xc000602900, 0xc002496000)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd

Error: The terraform-provider-azurerm_v2.66.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2021-07-05T16:47:22.699+0200 [DEBUG] provider: plugin exited
```

--- a/azurerm/internal/services/recoveryservices/site_recovery_replicated_vm_resource.go
+++ b/azurerm/internal/services/recoveryservices/site_recovery_replicated_vm_resource.go
@@ -457,12 +457,41 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
 			disksOutput := make([]interface{}, 0)
 			for _, disk := range *a2aDetails.ProtectedManagedDisks {
 				diskOutput := make(map[string]interface{})
-				diskOutput["disk_id"] = *disk.DiskID
-				diskOutput["staging_storage_account_id"] = *disk.PrimaryStagingAzureStorageAccountID
-				diskOutput["target_resource_group_id"] = *disk.RecoveryResourceGroupID
-				diskOutput["target_replica_disk_type"] = *disk.RecoveryReplicaDiskAccountType
-				diskOutput["target_disk_type"] = *disk.RecoveryTargetDiskAccountType
-				diskOutput["target_disk_encryption_set_id"] = *disk.RecoveryDiskEncryptionSetID
+				diskId := ""
+				if disk.DiskID != nil {
+					diskId = *disk.DiskID
+				}
+				diskOutput["disk_id"] = diskId
+
+				primaryStagingAzureStorageAccountID := ""
+				if disk.PrimaryStagingAzureStorageAccountID != nil {
+					primaryStagingAzureStorageAccountID = *disk.PrimaryStagingAzureStorageAccountID
+				}
+				diskOutput["staging_storage_account_id"] = primaryStagingAzureStorageAccountID
+
+				recoveryResourceGroupID := ""
+				if disk.RecoveryResourceGroupID != nil {
+					recoveryResourceGroupID = *disk.RecoveryResourceGroupID
+				}
+				diskOutput["target_resource_group_id"] = recoveryResourceGroupID
+
+				recoveryReplicaDiskAccountType := ""
+				if disk.RecoveryReplicaDiskAccountType != nil {
+					recoveryReplicaDiskAccountType = *disk.RecoveryReplicaDiskAccountType
+				}
+				diskOutput["target_replica_disk_type"] = recoveryReplicaDiskAccountType
+
+				recoveryTargetDiskAccountType := ""
+				if disk.RecoveryTargetDiskAccountType != nil {
+					recoveryTargetDiskAccountType = *disk.RecoveryTargetDiskAccountType
+				}
+				diskOutput["target_disk_type"] = recoveryTargetDiskAccountType
+
+				recoveryEncryptionSetId := ""
+				if disk.RecoveryDiskEncryptionSetID != nil {
+					recoveryEncryptionSetId = *disk.RecoveryDiskEncryptionSetID
+				}
+				diskOutput["target_disk_encryption_set_id"] = recoveryEncryptionSetId
 
 				disksOutput = append(disksOutput, diskOutput)
 			}

